components/script/dom/webglcontextevent.rs

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::cell::DOMRefCell;
+use dom::bindings::codegen::Bindings::EventBinding::EventMethods;
+use dom::bindings::codegen::Bindings::WebGLContextEventBinding;
+use dom::bindings::codegen::Bindings::WebGLContextEventBinding::WebGLContextEventInit;
+use dom::bindings::codegen::Bindings::WebGLContextEventBinding::WebGLContextEventMethods;
+use dom::bindings::codegen::InheritTypes::{WebGLContextEventDerived, EventCast};
+use dom::bindings::error::Fallible;
+use dom::bindings::global::GlobalRef;
+use dom::bindings::js::{Root, RootedReference};
+use dom::bindings::utils::reflect_dom_object;
+use dom::event::{Event, EventBubbles, EventCancelable, EventTypeId};
+use util::str::DOMString;
+
+#[dom_struct]
+pub struct WebGLContextEvent {
+    event: Event,
+    status_message: DOMRefCell<DOMString>,
+}
+
+impl WebGLContextEventMethods for WebGLContextEvent {
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.15
+    fn StatusMessage(&self) -> DOMString {
+        self.status_message.borrow().clone()
+    }
+}
+
+impl WebGLContextEventDerived for Event {
+    fn is_webglcontextevent(&self) -> bool {
+        *self.type_id() == EventTypeId::WebGLContextEvent
+    }
+}
+
+impl WebGLContextEvent {
+    pub fn new_inherited(type_id: EventTypeId) -> WebGLContextEvent {
+        WebGLContextEvent {
+            event: Event::new_inherited(type_id),
+            status_message: DOMRefCell::new("".to_owned()),
+        }
+    }
+
+    pub fn new_uninitialized(global: GlobalRef) -> Root<WebGLContextEvent> {
```

Comment:
This can be removed like in `storageevent`, but I wasn't sure about what was the bes option, so I've kept it.